### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by RimSebai**

### CODE REVIEW

File: Week3/App.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    const article = Util.createAndAppend("article", root, { id: "container" });
+    Util.createAndAppend("p", header, { text: "HYF  Repositories", id: "HYF_Repositories" });
+    const select = Util.createAndAppend("select", header);
+
+    try {
+
+      const repos = await Util.fetchJSON(url);
+      this.repos = repos.sort((a, b) => a.name.localeCompare(b.name)).map(repo => new Repository(repo));
+      for (let i = 0; i < this.repos.length; i++) {
+        Util.createAndAppend("option", select, { text: this.repos[i].data.name, value: i, class: "options" });
```
Comment (remarcmij) [inline Week3/App.js]: As the `Repository` object has a `.name()` method you could also code this as:

```js
Util.createAndAppend("option", select, { text: this.repos[i].name(), value: i, class: "options" });
```

File: Week3/App.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    const root = document.getElementById('root');
+    const header = Util.createAndAppend("header", root, { id: "mainCon" });
+    const article = Util.createAndAppend("article", root, { id: "container" });
+    Util.createAndAppend("p", header, { text: "HYF  Repositories", id: "HYF_Repositories" });
+    const select = Util.createAndAppend("select", header);
+
+    try {
+
+      const repos = await Util.fetchJSON(url);
+      this.repos = repos.sort((a, b) => a.name.localeCompare(b.name)).map(repo => new Repository(repo));
```
Comment (remarcmij) [inline Week3/App.js]: It is customary to split a method chain over multiple lines if the line becomes long, e.g.:

```js
this.repos = repos
  .sort((a, b) => a.name.localeCompare(b.name))
  .map(repo => new Repository(repo));
```

File: Week3/Contributor.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(data) {
+    this.data = data;
+  }
+
+  /**
+   * Render the contributor info to the DOM.
+   * @param {HTMLElement} contributorList The parent element in which to render the contributor.
+  */
+  render(contributorList) {
+    const rightDiv = Util.createAndAppend("div", contributorList, { id: "contributors" });
```
[the remark this answers is not in the document]
Comment (RimSebai) [reply Week3/Contributor.js]: so the div is enough?

[the remark this answers is not in the document]
Comment (remarcmij) [reply Week3/Contributor.js]: No, the `contributorList` parameter of this function is actually a `ul` (see line 67 of `App.js`). So each contributor should be embedded within an `li`. For instance:

```html
<ul> <!-- this is contributorList that gets passed to your function
  <li id="contributor">
    <img src="https://avatars1.githubusercontent.com/u/37488847?v=4" alt="wouterkleijn's img">
    <p id="name">
       <a href="https://github.com/wouterkleijn" target="_blank" id="links">wouterkleijn</a>
    </p>
    <div id="contributions">4</div>
  </li>
   <!-- more contributors here, one 'li' per contributor -->
</ul>

**Pull request closed without merging**